Chall type separation

v1bh475u opened the pull request:
closes #408

v1bh475u committed:
fix: update errors as per convention

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
feat: add field for service config file name

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
mod: remove DOCKER CHALLENGE TYPE

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
feat: make servicefile and dockerfile optional fields

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
mod: update mod and sum files

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
refactor

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
mod: rename service_config to xinetd_conf

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
refactor

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
feat: add XinetdConf field to BeastBareDockerfile and update entrypoint script

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
fix(stageChallenge): correct condition for Dockerfile deletion

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
update(static-chall): add static_dir to challenge_env

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
add(xinetd-service): sample chal with user xinetd.conf

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
add(service): sample chal without author provided xinetd.conf

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
update(dependencies): upgrade gorm and sqlite driver versions

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

sukhman-sukh commented on core/manager/challenge.go:
```diff
@@ -440,13 +439,16 @@ func HandleAll(action string, user string) []string {
-		challsNameList, err := GetAvailableChallenges()
-		if err != nil || len(challsNameList) == 0 {
+		var deployChalls []string
+		deployChalls, err = GetAvailableChallenges()
```
Why this ?

sukhman-sukh commented on core/manager/challenge.go:
```diff
@@ -440,13 +439,16 @@ func HandleAll(action string, user string) []string {
 
 	case core.MANAGE_ACTION_UNDEPLOY:
-		challenges, err := database.QueryChallengeEntriesMap(map[string]interface{}{
+		var challenges []database.Challenge
```
Again, what was the issue with :=

v1bh475u replied:
I don't remember it that well but `:=` was creating `switch-scope` local variables which were getting populated instead of the function-scoped variables. So, basically, no error was getting propagated.

sukhman-sukh replied:
But you have again declared inside switch-scope. 
And I highly doubt there is any difference between the two,  given declared in same scope

v1bh475u replied:
The `err` variable is not declared in `switch-scope`

sukhman-sukh replied:
ok
But change the other ones where it is not required

sukhman-sukh replied:
And resolve them .

sukhman-sukh commented on core/manager/challenge.go:
```diff
@@ -456,7 +458,8 @@ func HandleAll(action string, user string) []string {
 
 	case core.MANAGE_ACTION_REDEPLOY:
-		challenges, err := database.QueryChallengeEntriesMap(map[string]interface{}{
+		var challenges []database.Challenge
```
same

sukhman-sukh commented on core/manager/challenge.go:
```diff
@@ -466,7 +469,8 @@ func HandleAll(action string, user string) []string {
 
 	case core.MANAGE_ACTION_PURGE:
-		challenges, err := database.QueryAllChallenges()
+		var challenges []database.Challenge
```
same

sukhman-sukh commented on core/manager/static.go:
```diff
@@ -39,6 +39,10 @@ func DeployStaticContentContainer() error {
 
 	images, err := cr.SearchImageByFilter(map[string]string{"reference": fmt.Sprintf("%s:latest", core.BEAST_STATIC_CONTAINER_NAME)})
+	if err != nil {
+		log.Errorf("Error searching for static content image: %s", err)
```
Any reason for change here ?

sukhman-sukh commented on core/manager/pipeline.go:
```diff
@@ -41,32 +41,45 @@ func stageChallenge(challengeDir string, config *cfg.BeastChallengeConfig) error
+	}
 	additionalCtx := make(map[string]string)
 	additionalCtx["Dockerfile"] = dockerfileCtx
+	additionalCtx[core.DEFAULT_XINETD_CONF_FILE] = serviceConfig
```
What if control never enters inside if block above and serviceConfig's value is not set?
Is that handled anywhere?

v1bh475u replied:
Yes.
https://github.com/sdslabs/beast/blob/51fb2a5bda779a71241a22027868dbc20451ee18/core/manager/utils.go#L355-L388

sukhman-sukh replied:
ok

sukhman-sukh commented on core/manager/challenge.go:
```diff
@@ -647,7 +651,7 @@ func undeployChallenge(challengeName string, purge bool) error {
 			// the container instance does not exist and hence should update the database
 			// with the container ID.
-			p := fmt.Errorf("Error while removing challenge instance : %s", err)
+			p := fmt.Errorf("error while removing challenge instance : %s", err)
```
Please keep formatting in seperate PR if there is a lot of formatting. 
It makes it hard to review.

v1bh475u committed:
fix(challenge-config): loosen up constraints on config file
- Bare : if dockerfile provided, other fields are optional
- Web: webroot not required if dockerfile provided

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
update: examples

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
update(README): add descriptions of new challenges

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
mod: update dependencies and fix bugs

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

sukhman-sukh commented on _examples/README.md:
```diff
@@ -13,7 +13,9 @@
-* [Xinted Service challenge](./xinetd-service)
+* [Xinted Service challenge with custom xinetd config](./xinetd-service)
+* [Service challenge with auto-generated xinetd config](./service)
+* [A bare challenge using docker](./docker-type)
```
Name is a bit confusing. Change docket-type to something like bare-docker just like xinet.d-service

v1bh475u replied:
shall I do the same for compose-type?

sukhman-sukh commented on api/response.go:
```diff
@@ -9,7 +9,7 @@ type HTTPPlainResp struct {
 
 type HTTPPlainMapResp struct {
-	Messages map[string]string `json:"messages" example:"["name1": "message1"],["name2": "message2"]"`
+	Messages map[string]string `json:"messages" example:"{\"name1\": \"message1\", \"name2\": \"message2\"}"`
```
Any reason for this?

v1bh475u replied:
unescaped string not working as intended

v1bh475u committed:
refactor: rename example challenge name for better understanding

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
fix: correct link for bare docker challenge in README

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
fix: correct typo in error log message for remote container search

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

v1bh475u committed:
fix: append challenge names to list in HandleAll function

Signed-off-by: vibhatsu <maulikbarot2915@gmail.com>

sukhman-sukh commented:
LGTM.

sukhman-sukh committed:
Merge branch 'bl4ze/dev' into chall-type-separation
